### PR TITLE
feat(server): Enterprise cron that rotates the current JWT signing key

### PR DESCRIPTION
**Pull request by charlesBochet**
## Summary
Adds a daily Enterprise-only cron that rotates the current ES256 JWT signing key once it has been current for `SIGNING_KEY_ROTATION_DAYS`. Manual rotation from the admin panel is unaffected.

### Behaviour
- `SIGNING_KEY_ROTATION_DAYS` is **opt-in**: when unset, the cron is a no-op.
- Rotation flips `isCurrent` and clears the previous key's `privateKey` in the same transaction, then inserts the new `isCurrent=true` row.
- The previous key's row is kept (`revokedAt` stays `null`) so its `publicKey` can keep verifying tokens it signed until they expire; only the encrypted `privateKey` is wiped since it can no longer be used to sign.
- **No auto-revocation** — revoking a key remains a manual admin action, reserved for leak / emergency response.
- The cron is also a no-op when `EnterprisePlanService.isValid()` is `false`.

### Wiring
- `JwtKeyManagerService.rotateCurrent()`
- `SigningKeyRotationService.rotateIfDue()` (reads `SIGNING_KEY_ROTATION_DAYS`, skips when unset)
- `RotateSigningKeysCronJob` (Enterprise-gated, rethrows on failure) registered in `JwtModule`
- `RotateSigningKeysCronCommand` registered with `cron:register:all`
- `ROTATE_SIGNING_KEYS_CRON_PATTERN = '15 3 * * *'` (daily, no-op until threshold)

Operator documentation lives in #20611 (docs PR).

### CODE REVIEW

File: packages/twenty-server/src/engine/core-modules/jwt/services/signing-key-rotation.service.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { JwtKeyManagerService } from 'src/engine/core-modules/jwt/services/jwt-key-manager.service';
+import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
+
+const ONE_DAY_MS = 24 * 60 * 60 * 1000;
+
+export type SigningKeyRotationResult = {
+  rotated: boolean;
+  previousId: string | null;
+  newId: string | null;
+};
```
Comment (prastoin) [inline packages/twenty-server/src/engine/core-modules/jwt/services/signing-key-rotation.service.ts:16]: ***Polish:*** Assuming we always have a previousId on first rotation
```ts
export type SigningKeyRotationResult = {
  rotated: true;
  previousId: string; // First rota could be null ?
  newId: string;
} | {
  rotated: false;
  previousId: never;
  newId: never;
};;
```

File: packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts [hunk 2]
```diff
@@ -78,6 +78,38 @@ export class JwtKeyManagerService {
...
+    const generated = this.generateEcP256KeyPair();
+    const newId = randomUUID();
+
+    await this.signingKeyRepository.manager.transaction(
+      async (entityManager) => {
+        const repository = entityManager.getRepository(SigningKeyEntity);
+
+        await repository.update(
+          { isCurrent: true },
+          { isCurrent: false, privateKey: null },
```
[the remark this answers is not in the document]
Comment (prastoin) [reply packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts:89]: +1

**Review by prastoin — COMMENTED** [review]
Critical issue with rotation service erasing previous private key

File: packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts [hunk 3]
```diff
@@ -78,6 +78,38 @@ export class JwtKeyManagerService {
...
+          ),
+          isCurrent: true,
+          revokedAt: null,
+        });
+      },
+    );
+
+    await this.coreEntityCacheService.invalidate('signingKeyPublicKey', newId);
+    this.currentSigningKeyPromise = null;
+    this.currentSigningKeyCachedAt = 0;
```
Comment (prastoin) [inline packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts]: Shouldn't this be now ?

File: packages/twenty-server/test/integration/graphql/suites/auth/jwt-key-rotation/rotate-signing-keys-cron-job.integration-spec.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    const rotatedKid = decodeJwtCompleteOrThrow(rotatedApiKeyToken).header
+      .kid as string;
+
+    expect(isNonEmptyString(rotatedKid)).toBe(true);
+    expect(rotatedKid).not.toBe(initialKid);
+
+    const callWithPreviousToken = await findManyApplications({
+      accessToken: initialApiKeyToken,
+      expectToFail: false,
+    });
```
Comment (prastoin) [inline packages/twenty-server/test/integration/graphql/suites/auth/jwt-key-rotation/rotate-signing-keys-cron-job.integration-spec.ts:92]: should be failing unless I'm mistaken

File: packages/twenty-server/src/engine/core-modules/jwt/services/signing-key-rotation.service.ts (hunk 1 above)
Comment (charlesBochet) [reply packages/twenty-server/src/engine/core-modules/jwt/services/signing-key-rotation.service.ts:16]: it could it some really specific edge case where the currentKey is revoked from admin panel and the cron is triggered. This is edge case, I can also make the service throw but both way looks OK

File: packages/twenty-server/test/integration/graphql/suites/auth/jwt-key-rotation/rotate-signing-keys-cron-job.integration-spec.ts (hunk 4 above)
Comment (charlesBochet) [reply packages/twenty-server/test/integration/graphql/suites/auth/jwt-key-rotation/rotate-signing-keys-cron-job.integration-spec.ts:92]: why? privateKey is nulled but we still have publicKey to verify tokens

File: packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts (hunk 3 above)
Comment (charlesBochet) [reply packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts]: this is marking the cache as expired:

In upgcoming calls:
 const isLocalCacheExpired =
      Date.now() - this.currentSigningKeyCachedAt >
      CURRENT_SIGNING_KEY_LOCAL_TTL_MS;
      
Putting 0 is to say that nothing is cached.

If we put Date.now(), this means we would have to also have to compute the cache which is not the case here

File: packages/twenty-server/test/integration/graphql/suites/auth/jwt-key-rotation/rotate-signing-keys-cron-job.integration-spec.ts (hunk 4 above)
Comment (prastoin) [reply packages/twenty-server/test/integration/graphql/suites/auth/jwt-key-rotation/rotate-signing-keys-cron-job.integration-spec.ts:92]: I meant regarding the previous bug introduced in the rotation erasing previous privateKEy

File: packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts (hunk 3 above)
Comment (prastoin) [reply packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts]: Might wanna create a dedicated method abstraction as it's quite implicit

Comment (charlesBochet) [reply packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts]: Ok I can do that but that's nitpick, this code is very local and not exposed so easy to refactor / maintain and not bug prone!

Comment (charlesBochet) [reply packages/twenty-server/src/engine/core-modules/jwt/services/jwt-key-manager.service.ts]: For you, I'm doing it :D
done

**Review by prastoin — APPROVED** [review]
Good !
